### ISSUE TITLE
Nginx

### ISSUE THREAD

**Issue by bcsanford** [bolkedebruin/rdpgw#30]
@bolkedebruin
can this be setup behind a nginx reverse proxy so that is can share the IP address with other websites, i know you can do this with mesh central, but the goals are slightly different

label:question

**Comment by krisss85** [comment]
@bcsanford we have it running behind nginx ingress controller on k8s so should be all fine

**Comment by chiemseesurfer** [comment]
Hey, I try to get that running behind ingress controller on k8s. Nginx should do the ssl part.
Authentication is set to ntlm and Tls is set to disable on rdpgw side.
As far as I can see connection from nginx to rdpgw works.
I  can see successful authentication for ntlm in rdpgw logs too. But after all I see 401 on nginx side:
`"RDG_OUT_DATA /remoteDesktopGateway/ HTTP/1.1"`.
If I'm right the first try is always 401?! At least if I try it without nginx and rdpgw directly (for sure TLS enabled) I see that 401 in remmina trace logs too.

Question now: What am I missin on nginx ingress side? Annotations are similar to that one from @krisss85 . Is there somewhere an example nginx ingress config available?

**Comment by firmanalams** [comment]
I am having the same issue. I have this `nginx -> rdpgw -> rdp server`. And here my nginx configuration

```
proxy_pass http://rdpgw:8081;
proxy_set_header Host $host;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header Upgrade "websocket";
proxy_set_header Connection "Upgrade";
```

But I got this error
`Cannot upgrade falling back to old protocol: {%!t(string=websocket: unsupported version: 13 not found in 'Sec-Websocket-Version' header)}`

Please if anyone has successfully used reverse proxy

**Comment by jothoma1** [comment]
Hi

Here is my working config for `nginx -> rdpgw -> rdp server`. 

```
server {
        listen 443 ssl http2;
        ssl_certificate /etc/ssl/certs/dddddd.crt;
        ssl_certificate_key /etc/ssl/private/dddddd.key;

        server_name my-great-public-rdpgw.domain.com;

        proxy_busy_buffers_size   512k;
        proxy_buffers  4 512k;
        proxy_buffer_size   256k;

        access_log /var/log/nginx/rdpgw.access.log json_combined;

        set_real_ip_from 0.0.0.0/0;
        real_ip_header X-Real-IP;
        real_ip_recursive on;

        location / {
           proxy_pass http://rdpgw.internal;

           proxy_redirect off;
           proxy_set_header Host $host;
           proxy_set_header X-Real-IP $remote_addr;
           proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
           proxy_set_header X-Forwarded-Proto $scheme;
           proxy_set_header X-Forwarded-Port 443;
           proxy_set_header X-Forwarded-Host $host:443;
           proxy_set_header X-Forwarded-Server $host;
           proxy_set_header Upgrade $http_upgrade;
           proxy_set_header Connection "Upgrade";
        }
}
```

**Comment by firmanalams** [comment]
@jothoma1 Thank you for your help. I have set my config like yours. But, I get a timeout error when requesting `RDG_OUT_DATA /remoteDesktopGateway/`. Am I missing something?

**Comment by jothoma1** [comment]
Just a precision, i'm using openid for authentication and i dont use Remmina

**Comment by firmanalams** [comment]
You are right. I successfully used `mstsc`. I think this problem lies in Remmina
